### PR TITLE
Pre-sort currency and language data

### PR DESCRIPTION
**Pull request by c960657**
`Currency::getAllCurrencies()` and `Language::getAll()` both filter a list of strings and them sort them. If the strings are sorted in advance, we don't have to do it at runtime.

In addition, we reduce the runtime dependency on the `Collator`class from the Intl extension.

### CODE REVIEW

File: bin/punic-build-data [hunk 1]
```diff
@@ -72,6 +74,9 @@ try {
         $sourceData->ensureJsonLocale('en');
         echo "done.\n";
     }
+    if (!class_exists('\Collator')) {
```
Comment (mlocati) [inline bin/punic-build-data]: @c960657 Is this required? Punic `Comparer` class should handle the case when it's missing, isn't it?

**Comment by mlocati** [comment]
JSON objects are unordered set of name/value pairs, so I'm not 100% sure we should rely on that...

File: bin/punic-build-data (hunk 1 above)
Comment (c960657) [reply bin/punic-build-data]: `Punic\Comparer` uses an approximated comparison method that is of lower quality than that in `Collator`.

Most users wont need to run this script, and they don't need to run in on their production server, so I think the `intl` extension is a modest requirement.

Comment (mlocati) [reply bin/punic-build-data]: I do agree

**Comment by c960657** [comment]
True, but I believe PHP preserves the order when encoding and decoding JSON. But I agree it's a bit ugly.

If you like the idea suggested in #103, we can avoid the JSON problem.

**Comment by mlocati** [comment]
I like both #103 and #107, but here's what I'd like:
- a  repository for the data (`punic/data`) which uses Docker to build the data for *all* the locales and for all the supported CLDR versions, both in JSON and PHP format
-  a CLI command in the main repository (`punic/punic`) that fetches the data from that repository; users will be able to choose the CLDR version, the data format (JSON or PHP), and the locales

**Comment by c960657** [comment]
What is the point in allowing two different data formats? The files are not consumed or generated by other software, so why would anybody not want to use the default format?

**Comment by mlocati** [comment]
Maybe someone would prefer JSON because he doesn't have opcache enabled and PHP would be slower in that case... but you are right: if someone cares about speed he should install opcache. So, let's switch to PHP only data files.

**Comment by mlocati** [comment]
PR rebased in #119

**Pull request closed without merging**